src/yggdrasil/nodeinfo.go

```diff
@@ -108,9 +108,14 @@ func (m *nodeinfo) setNodeInfo(given interface{}) error {
 	if nodeinfomap, ok := given.(map[string]interface{}); ok {
```

Comment:
unrelated, but if `given` is always expected to be of type `map[string]interface{}`, maybe it should be just cast without ok check

Reply:
It can be `null`.